salt/modules/win_pki.py

```diff
@@ -155,7 +155,7 @@ def get_certs(context=_DEFAULT_CONTEXT, store=_DEFAULT_STORE):
             if key not in blacklist_keys:
                 cert_info[key.lower()] = item[key]
 
-        cert_info['dnsnames'] = [name['Unicode'] for name in item['DnsNameList']]
+        cert_info['dnsnames'] = [name['Unicode'] for name in item['DnsNameList'] or []]
```

Comment:
@twangboy `name['Unicode']` expects `name` to be a dict, while it falls back to a list in your suggestion. If my original patch is lacking in readability, I could just change it to a regular if/else block?